x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/action/PostAnalyticsEventAction.java

```diff
@@ -0,0 +1,248 @@
+        }
+
+        public long eventTime() {
+            return eventTime;
+        }
+
+        public BytesReference payload() {
+            return payload;
+        }
+
+        public boolean isDebug() {
+            return debug;
+        }
+
+        public XContentType xContentType() {
+            return xContentType;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+
+            if (Strings.isNullOrEmpty(eventCollectionName)) {
+                validationException = addValidationError("collection name is missing", validationException);
+            }
+
+            if (Strings.isNullOrEmpty(eventType)) {
+                validationException = addValidationError("event type is missing", validationException);
+            }
+
+            try {
+                if (eventType.toLowerCase(Locale.ROOT).equals(eventType) == false) throw new IllegalArgumentException();
+
+                AnalyticsEvent.Type.valueOf(eventType.toUpperCase(Locale.ROOT));
+            } catch (IllegalArgumentException e) {
+                validationException = addValidationError(
+                    LoggerMessageFormat.format("invalid event type: {}", eventType),
+                    validationException
+                );
+            }
+
+            return validationException;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeString(eventCollectionName);
+            out.writeString(eventType);
+            out.writeBoolean(debug);
+            out.writeLong(eventTime);
+            XContentHelper.writeTo(out, xContentType);
+            out.writeBytesReference(payload);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Request that = (Request) o;
+            return Objects.equals(eventCollectionName, that.eventCollectionName)
+                && debug == that.debug
+                && eventTime == that.eventTime
+                && Objects.equals(eventType, that.eventType)
+                && Objects.equals(xContentType, that.xContentType)
+                && Objects.equals(payload, that.payload);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(eventCollectionName, eventType, debug, eventTime, xContentType, payload);
+        }
+    }
+
+    public static class Response extends ActionResponse implements ToXContentObject {
+
+        public static Response ACCEPTED = new Response(true);
+
+        private static final ParseField RESULT_FIELD = new ParseField("accepted");
+
+        private static final ParseField EVENT_FIELD = new ParseField("event");
+
+        private final boolean accepted;
+
+        private final boolean debug;
```

Comment:
Nitpick: Would a variable name such as `verbose` or `showEvent` be more clear than `debug`?